tests/integration/restricted/test.sh

```diff
@@ -44,51 +44,60 @@ kubectl apply -f sourcegraph.StorageClass.yaml
 
 kubectl apply -f nonroot-policy.yaml
 
-kubectl create namespace ns-sourcegraph
+export NAMESPACE=$CLUSTER_NAME
+
+kubectl create namespace $NAMESPACE
 
 # Deleting the namespace during cleanup is a cheap way to delete associated PVC's - see https://issuetracker.google.com/issues/121034250?pli=1
-CLEANUP="kubectl delete namespace ns-sourcegraph --timeout=180s; $CLEANUP"
+CLEANUP="kubectl delete namespace $NAMESPACE --timeout=180s; $CLEANUP"
 
-kubectl create serviceaccount -n ns-sourcegraph fake-user
+kubectl create serviceaccount -n $NAMESPACE fake-user
 
-kubectl create rolebinding -n ns-sourcegraph fake-admin --clusterrole=admin --serviceaccount=ns-sourcegraph:fake-user
+kubectl create rolebinding -n $NAMESPACE fake-admin --clusterrole=admin --serviceaccount=$NAMESPACE:fake-user
 
 # Kubernetes < 1.16 change to '--resource=podsecuritypolicies.extensions'
-kubectl create role -n ns-sourcegraph nonroot:unprivileged --verb=use --resource=podsecuritypolicies.extensions --resource-name=nonroot-policy
+kubectl create role -n $NAMESPACE nonroot:unprivileged --verb=use --resource=podsecuritypolicies.extensions --resource-name=nonroot-policy
 
-kubectl create rolebinding -n ns-sourcegraph fake-user:nonroot:unprivileged --role=nonroot:unprivileged --serviceaccount=ns-sourcegraph:fake-user
+kubectl create rolebinding -n $NAMESPACE fake-user:nonroot:unprivileged --role=nonroot:unprivileged --serviceaccount=$NAMESPACE:fake-user
 
 /bin/cat <<EOM >deploy_sourcegraph_git_ssh_config
 Host *
     StrictHostKeyChecking no
 EOM
 
-kubectl create secret -n ns-sourcegraph generic gitserver-ssh --from-literal=rsa=supersecret --from-literal=config=topsecret
+kubectl create secret -n $NAMESPACE generic gitserver-ssh --from-literal=rsa=supersecret --from-literal=config=topsecret
 
 mkdir generated-cluster
 CLEANUP="rm -rf generated-cluster; $CLEANUP"
+sed -i "s/ns-sourcegraph/$NAMESPACE/" "${DEPLOY_SOURCEGRAPH_ROOT}/overlays/non-privileged-create-cluster/kustomization.yaml"
+
 "${DEPLOY_SOURCEGRAPH_ROOT}"/overlay-generate-cluster.sh non-privileged-create-cluster ${CURRENT_DIR}/generated-cluster
 
+# TODO There must be a better way to do this
```

Comment:
@bobheadxi that fixed it, I wonder why the kustomize.yml change didn't apply there 🤔

Reply:
This could be something we can solve better with Helm. I think a workaround is fine for now - I wonder if eventually we should move these comprehensive tests over to https://github.com/sourcegraph/deploy-sourcegraph-helm and effectively deprecate the integration testing done here